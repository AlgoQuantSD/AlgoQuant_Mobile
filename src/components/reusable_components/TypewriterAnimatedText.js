import React, { useState, useEffect } from "react";
import { View, Text } from "react-native";

export default function TypewriterAnimatedText(props) {
  const { text, style } = props;
  const [typedText, setTypedText] = useState("");
  const textToType = text;

  useEffect(() => {
    let index = 0;
    const intervalId = setInterval(() => {
      setTypedText(textToType.slice(0, index + 1));
      index++;
      if (index === textToType.length) {
        clearInterval(intervalId);
      }
    }, 100);
    return () => clearInterval(intervalId);
  }, [textToType]);

  return (
    <View>
      <Text style={style}>{typedText}</Text>
    </View>
  );
}
