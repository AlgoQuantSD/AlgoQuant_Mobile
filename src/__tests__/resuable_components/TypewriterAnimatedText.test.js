import React from "react";
import { render, waitFor } from "@testing-library/react-native";
import TypewriterAnimatedText from "../../components/reusable_components/TypewriterAnimatedText";

describe("TypewriterAnimatedText", () => {
  it("should render", () => {
    const tree = render(<TypewriterAnimatedText />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should type out the text over time", async () => {
    const text = "Hello World";
    const { getByText } = render(<TypewriterAnimatedText text={text} />);

    expect(getByText("")).toBeTruthy();

    await waitFor(() => {
      expect(getByText("H")).toBeTruthy();
    });

    await waitFor(() => {
      expect(getByText("He")).toBeTruthy();
    });

    await waitFor(() => {
      expect(getByText("Hello World")).toBeTruthy();
    });
  });
});
