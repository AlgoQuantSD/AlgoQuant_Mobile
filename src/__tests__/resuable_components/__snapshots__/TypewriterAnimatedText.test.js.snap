// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypewriterAnimatedText should render 1`] = `
<View>
  <Text />
</View>
`;
